backend/strapi-deploy

```diff
@@ -1,14 +1,37 @@
 #!/bin/bash
 
-set -o pipefail -e
+set -o pipefail -e nounset
 
 export STRAPI_COMMIT=$(git rev-parse HEAD)
+DEPLOY_TAG=last-strapi-deploy
 
-dir=$(git rev-parse --show-toplevel)
-cd "$dir/backend"
+git fetch --tags --quiet
+
+# Short circuit if the build branch is not main.
+DEPLOY_BRANCH="main"
+if [[ $BUILD_BRANCH != $DEPLOY_BRANCH ]]; then
+   echo "Skipping $BUILD_BRANCH. We only deploy $DEPLOY_BRANCH.";
+   exit 0
+fi
+
+echo "Previous deploy was $(git rev-parse "$DEPLOY_TAG"), listing backend/ changes:"
+if git diff "$DEPLOY_TAG...HEAD" --exit-code --name-only -- backend/; then
```

Comment:
This logic seems backwards. If there are _are_ changes to `backend/`, then `git diff` will exit with `1`, and this if statement will exit the script early.
```suggestion
if ! git diff "$DEPLOY_TAG...HEAD" --exit-code --name-only -- backend/; then
```

Reply:
We get an exit 0 when there is no diff and 1 when there *is* a diff. As backward as it seems, that's the way diff (and git diff) works:

![image](https://user-images.githubusercontent.com/26855/232598827-80ff6766-5be3-46cd-a5f5-770265d66989.png)

Reply:
Derp, I was confused that a `0` exit code is truthy in bash and an exit code of `1` is falsy. Reading it this way, it now looks right 👍🏻